### ISSUE TITLE
Start to train model and it stuck

### ISSUE THREAD

**Issue by AGanPoi** [open-mmlab/mmdetection#6169]
When start to train model，the volatile GPU-Util come to 100% and the console do not throw any information as if it is stuck.
![2](https://user-images.githubusercontent.com/35792267/134862853-32a98d32-809f-4477-9df7-87ec6284d124.png)

My GPU is V100,  cuda and nvcc version is 10.1
![1](https://user-images.githubusercontent.com/35792267/134862971-6962e574-e6bc-406f-8592-17f36bf22c33.png)

This is my environmental information, it seem very correct and I cannot find any error
![3](https://user-images.githubusercontent.com/35792267/134863103-383f0f59-97b9-4feb-90df-dfc32c017924.png)

**Comment by AGanPoi** [comment]
And I press "Ctrl+C" to stop it, the volatile GPU-Util is also 100% and the processes still alive.

**Comment by AGanPoi** [comment]
It seem that the distributed train in muti-GPU have error and could not stop directly
![QQ图片20210928083816](https://user-images.githubusercontent.com/35792267/135003740-8b9b6225-82e3-4b21-8b81-d4aa019deaf0.png)

**Comment by jshilong** [comment]
Which algorithm did you use and Is there any modification in the code?
- Usually, this is because you write some operations that need to synchronize between different ranks but some ranks can not reach the corresponding code line. 
- Another case is in your forward process, not all parameters participate in generating the loss, which makes some parameters do not have grad after loss.backward() which make reduce grad hanging

**Comment by AGanPoi** [comment]
> Which algorithm did you use and Is there any modification in the code?
> 
> * Usually, this is because you write some operations that need to synchronize between different ranks but some ranks can not reach the corresponding code line.
> * Another case is in your forward process, not all parameters participate in generating the loss, which makes some parameters do not have grad after loss.backward() which make reduce grad hanging

I just run the code of CVPR2021 paper “RefineMask：Towards High-Quality Instance Segmentation with Fine-Grained Features” and “Look Closer to Segment Better：Boundary Patch Refinement for Instance Segmentation”, they based on **_mmdetection_** and _**mmsegmentation**_, respectively. And they all come across this error. So I think is not the problem of code, maybe it's something wrong with my GPU and cuda.

Right now I just use another machine that is Tesla P100, and it run very well. I don't know why Tesla V100 cannot run in multi-GPU

**Comment by jshilong** [comment]
> > Which algorithm did you use and Is there any modification in the code?
> > 
> > * Usually, this is because you write some operations that need to synchronize between different ranks but some ranks can not reach the corresponding code line.
> > * Another case is in your forward process, not all parameters participate in generating the loss, which makes some parameters do not have grad after loss.backward() which make reduce grad hanging
> 
> I just run the code of CVPR2021 paper “RefineMask：Towards High-Quality Instance Segmentation with Fine-Grained Features” and “Look Closer to Segment Better：Boundary Patch Refinement for Instance Segmentation”, they based on **_mmdetection_** and _**mmsegmentation**_, respectively. And they all come across this error. So I think is not the problem of code, maybe it's something wrong with my GPU and cuda.
> 
> Right now I just use another machine that is Tesla P100, and it run very well. I don't know why Tesla V100 cannot run in multi-GPU

Are you sure all settings in the two devices identical?

**Comment by AGanPoi** [comment]
> > > Which algorithm did you use and Is there any modification in the code?
> > > 
> > > * Usually, this is because you write some operations that need to synchronize between different ranks but some ranks can not reach the corresponding code line.
> > > * Another case is in your forward process, not all parameters participate in generating the loss, which makes some parameters do not have grad after loss.backward() which make reduce grad hanging
> > 
> > 
> > I just run the code of CVPR2021 paper “RefineMask：Towards High-Quality Instance Segmentation with Fine-Grained Features” and “Look Closer to Segment Better：Boundary Patch Refinement for Instance Segmentation”, they based on **_mmdetection_** and _**mmsegmentation**_, respectively. And they all come across this error. So I think is not the problem of code, maybe it's something wrong with my GPU and cuda.
> > Right now I just use another machine that is Tesla P100, and it run very well. I don't know why Tesla V100 cannot run in multi-GPU
> 
> Are you sure all settings in the two devices identical?

If the environmental settings in mmcv and mmdet, or such as python version and pytorch version, that is the same beacuse I just copy the environmental settings. I found that is the problem of my GPU Tesla V100, the model can't be train on GPU id 0 and 1 simultaneously. 
Thank you for your reply and now my code is run well in other id of GPU device. I will refer to the Nvidia and NCLL to see how to slove this weird prolem.
